### PR TITLE
Improve select * from table limit 10 in FE

### PR DESCRIPTION
**Pull request by kangkaisen**
For https://github.com/StarRocks/starrocks/pull/1649

For SQL select * from table limit x, we could only query a few of tablets

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/DistributionPruneRule.java [hunk 1]
```diff
@@ -52,7 +53,8 @@ public DistributionPruneRule() {
         }
 
         // prune hint tablet
-        if (null != olapScanOperator.getHintsTabletIds() && !olapScanOperator.getHintsTabletIds().isEmpty()) {
+        Preconditions.checkState(olapScanOperator.getHintsTabletIds() != null);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/DistributionPruneRule.java:56]: Why check hint tablet？It's maybe null

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/LimitPruneTabletsRule.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        super(RuleType.TF_LIMIT_TABLETS_PRUNE, Pattern.create(OperatorType.LOGICAL_OLAP_SCAN));
+    }
+
+    private static final LimitPruneTabletsRule instance = new LimitPruneTabletsRule();
+
+    public static LimitPruneTabletsRule getInstance() {
+        return instance;
+    }
+
+    public boolean check(final OptExpression input, OptimizerContext context) {
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/LimitPruneTabletsRule.java:33]: is that a override method?

**Review by dirtysalt — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/DistributionPruneRule.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/DistributionPruneRule.java:56]: Yes.  I add code here

```
 private List<Long> tabletIds = Lists.newArrayList();
```

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/LimitPruneTabletsRule.java (hunk 2 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/LimitPruneTabletsRule.java:34]: Add     @Override

**Review by dirtysalt — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/DistributionPruneRule.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/DistributionPruneRule.java:56]: But the code in `TableRef`, not `LogicalOlapScanOperator`,  construct method of `LogicalOlapScanOperator` can't sure `hintsTabletIds` always not null, and I find many call without set `hintsTabletIds` in UT. Maybe we need add default value in `LogicalOlapScanOperator` or add @TODO to deal it later ?

**Review by Seaven — APPROVED** [review]

**Pull request merged**